### ISSUE TITLE
A -Depth parameter would be usefull

### ISSUE THREAD

**Issue by x86Since8088** [cloudbase/powershell-yaml#81]
I would like to dump data for human review in YAML, but some objects have seemingly endless recursion.  A -Depth parameter would help here.

Exception calling "Serialize" with "2" argument(s): "Too much recursion when traversing the object graph"
At C:\Users\my\Documents\WindowsPowerShell\Modules\powershell-yaml\0.4.2\powershell-yaml.psm1:398 char:13
+             $serializer.Serialize($wrt, $norm)
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : InvalidOperationException

**Comment by Serpen** [comment]
+1 for that, even standard thing fails quite quick

```powershell
gi C:\Windows\ | ConvertTo-Yaml
```

**Comment by gabriel-samfira** [comment]
note, the absolute max recursion is 1000. Default recursion is 100. Most yaml won't ever get close to 100 let alone 1000, but I have learned not to doubt human creativity. To be fair, the stable version is limited at 50 as it is, so these values should be fine.

**Comment by gabriel-samfira** [comment]
The PR was merged. There is now a `-Depth` flag for `ConvertTo-Yaml`

**Comment by RokeJulianLockhart** [comment]
@gabriel-samfira, is `-Depth` also limited (to 1000)?

**Comment by gabriel-samfira** [comment]
yes. The default is 100 (which is higher than the old default), but you can raise it up to 1000 if you need to.

**Comment by RokeJulianLockhart** [comment]
@gabriel-samfira, why does such a limit exist?

**Comment by gabriel-samfira** [comment]
There are self referencing, serializable objects like the `Root` object in the `Get-Item .` command you see above. Without a depth limiter, you will consume CPU forever, recursing to infinity. `YamlDotNet` itself has a limit of 50 by default. We raised that a bit and added the flag to the commandlet to allow a customizable range up to 1000. It is unlikely you will ever need more. If there are more than 1000 levels to a YAML, then YAML is probably the wrong tool for the job.
